src/Functions/array/arrayAUC.cpp

```diff
@@ -116,20 +125,86 @@ namespace ErrorCodes
   *   which uses linear interpolation and can be too optimistic for the Precision Recall AUC metric.
   */
 
-template <bool PR>
+template <bool is_pr>
 class FunctionArrayAUC : public IFunction
 {
 public:
-    static constexpr auto name = PR ? "arrayAUCPR" : "arrayROCAUC";
-    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionArrayAUC<PR>>(); }
+    static constexpr auto name = is_pr ? "arrayPRAUC" : "arrayROCAUC";
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionArrayAUC<is_pr>>(); }
 
 private:
+    static constexpr size_t partial_auc_offsets_arg_index = is_pr ? 2 : 3;
+    static constexpr size_t partial_auc_offsets_size = is_pr ? 3 : 4;
+
+    static Float64 increase_unscaled_area(size_t prev_fp, size_t prev_tp, size_t curr_fp, size_t curr_tp)
+    {
+        if constexpr (is_pr)
+            /** PR curve plots Precision x Recall
+              *
+              * Precision = TP / (TP + FP)
+              * Recall = TP / (TP + FN)
+              *
+              * The AUC is calculated using the Right Riemann Sum.
+              *
+              * Instead of calculating
+              *   area += (Recall_n - Recall_{n-1}) * Precision_n,
+              * we simplify it to
+              *   area += (TP_n - TP_{n-1}) * Precision_n
+              * and compute the "unscaled" area.
+              *
+              * The unscaled area represents the AUC of the Precision x TP curve.
+              * Later we can divide it by (TP + FN) to obtain the correct AUC.
+              *
+              * This can be done because (TP + FN) is constant and equal to total positive labels.
+              */
+            return static_cast<Float64>(curr_tp) / (curr_tp + curr_fp) * (curr_tp - prev_tp);
```

Comment:
Would you double-check that there cannot be a division by 0 in l. 160, 181, 190 and 195?